packages/contracts/contracts/authority/AuthorityFactory.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.13;
+
+import {Authority} from "@cartesi/rollups/contracts/consensus/authority/Authority.sol";
+import {History} from "@cartesi/rollups/contracts/history/History.sol";
+import {Create2} from "@openzeppelin/contracts/utils/Create2.sol";
+
+import {IAuthorityFactory} from "./IAuthorityFactory.sol";
+
+/// @title Authority-History Pair Factory
+/// @notice Allows anyone to reliably deploy a new Authority-History pair.
+contract AuthorityFactory is IAuthorityFactory {
+    function newAuthority(
+        address _authorityOwner
+    ) external override returns (Authority) {
+        Authority authority = new Authority(address(this));
+        History history = new History(address(authority));
+        authority.setHistory(history);
+        authority.transferOwnership(_authorityOwner);
+
+        emit AuthorityCreated(authority);
+
+        return authority;
+    }
+
+    function newAuthority() external override returns (Authority) {
+        return this.newAuthority(msg.sender);
+    }
+
+    function newAuthority(
+        address _authorityOwner,
+        bytes32 _salt
+    ) external override returns (Authority) {
+        _salt = calculateSalt(_authorityOwner, _salt);
+        Authority authority = new Authority{salt: _salt}(address(this));
+        History history = new History{salt: _salt}(address(authority));
+        authority.setHistory(history);
+        authority.transferOwnership(_authorityOwner);
+
+        emit AuthorityCreated(authority);
+
+        return authority;
+    }
+
+    function newAuthority(bytes32 _salt) external override returns (Authority) {
+        return this.newAuthority(msg.sender, _salt);
```

Comment:
Same here.

Reply:
But the function I'm calling is already a public function.
I just added this as a convenience option to remove one parameter.

Reply:
By public, I mean it has the `public` modifier in its signature.
All `newAuthority` functions in this contract are external.
External functions can only be called externally (`CALL`).
Public functions can be called externally (`CALL`) and internally (`JUMP`).

Reply:
So all `newAuthority` should be `public` instead of `external`?

Reply:
Not all. Only those you're calling internally.

Reply:
Done

Reply:
Now, you may remove the `this.` prefix.

```suggestion
        return newAuthority(msg.sender, _salt);
```